### ISSUE TITLE
Daylight Saving Accommodations Question?

### ISSUE THREAD

**Issue by henryblackman** [katzer/cordova-plugin-local-notifications#1241]
Hello,

It's not clear from the documentation, but do we need to concern ourselves with the upcoming change from winter time to daylight savings in most parts of the world.  I use notifications to reminder app users of their schedule, but do I need to adjust the time in which I schedule them by 1 hour forward?  For example (I'm in the UK so GMT/UTC time +0 hours currently) if I want a notification at 9am, do I schedule for 9am and expect it to be triggered at 9am local time or do I schedule for 8am and expect it to be triggered at 8am GMT/UTC which is actually 9am daylight savings (British Summer Time)?

I ask because the Javascript Date object, I understand (but I could be wrong) stores the time in Unix time*1000.  I'm reticent to use local notifications at all unless I've got clarity as I've got 20k users who would receive multiple notifications a day, I don't want to be wrong!

Any help would be appreciated.

**Comment by Tawpie** [comment]
The host device’s OS takes care of time zone and savings time changes… you schedule it to fire at 0800 and the device OS will fire the notification when the device’s time reaches 0800 (plus or minus for Android). This happens automatically. Remember that the device time can be set by the user to whatever they desire—so when things aren’t working as expected it’s a good start to check the device’s time.

**Comment by henryblackman** [comment]
Phew.  Thanks.  I had visions of telling 20k people to go a meeting an hour late ;-)

**Comment by henryblackman** [comment]
After some testing, I've found the OS does NOT account for daylight saving.

Fundamentally, creating a notification requires passing a Date object.  The Date object in Javascript is based on unix time, and as such is precisely the number of seconds from 1st Jan 1970.  So for a time scheduled, for example at 10:30am will when moving from non-daylight saving time to daylight saving time, will mean the notification is triggered one hour late.

To resolve this, I've used moment-timezone to create the Date object and ensuring I include the timezone (e.g. Europe/London), and it automatically accounts for the change to daylight saving.  Now - it seems - that I can set an appropriately timed notification, which I've tested by rolling time time forward to after the time shift and it is triggered correctly.
